### PR TITLE
Fix/mark required fields

### PR DESCRIPTION
**Pull request by tessi**
mark required fields and refactor some forms by the way

Fixes the following work packages:
- [User Story #2153](https://www.openproject.org/work_packages/2153): [Accessibility] Required fields MUST be displayed as required - group new
- [User Story #2157](https://www.openproject.org/work_packages/2157): [Accessibility] Required fields MUST be displayed as required - enumeration new
- [User Story #2162](https://www.openproject.org/work_packages/2162): [Accessibility] Required fields MUST be displayed as required - new project_type

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -116,6 +116,11 @@
   match '/help/:ctrl/:page' => 'help#index'
 
   resources :types
+  resources :statuses, :except => :show do
+    collection do
+      post 'update_work_package_done_ratio'
```
Comment (ulferts) [inline config/routes.rb:121]: Hoorayyyy for adding specific routes, booooh for not providing specs for those.

**Comment by ulferts** [comment]
As #788 is merged now, would you be so kind to apply your helpful required_field_name helper to 

app/views/planning_element_type_colors/_form.html.erb ?

Not a necessity but it would clean up the code.

File: app/views/account/register.html.erb [hunk 2]
```diff
@@ -31,52 +31,40 @@ See doc/COPYRIGHT.rdoc for more details.
```
Comment (sschu) [inline app/views/account/register.html.erb:31]: I dont want to spoil the party, but this view is overwritten by mobile_otp and self_registration_by_mail. So any changes here should also be done in these plugins.

File: app/views/auth_sources/_form.html.erb [hunk 3]
```diff
@@ -27,16 +27,13 @@ See doc/COPYRIGHT.rdoc for more details.
```
Comment (sschu) [inline app/views/auth_sources/_form.html.erb:27]: Its me again, the party crasher. auth_sources views are also contained in dtag_authentication - so any changes here should also be included there.

File: app/views/my/password.html.erb [hunk 4]
```diff
@@ -42,16 +42,14 @@ See doc/COPYRIGHT.rdoc for more details.
     </p>
```
Comment (sschu) [inline app/views/my/password.html.erb:42]: Once more, this one is overwritten by dtag_authentication.

**Comment by tessi** [comment]
@sschu in general I'd say that the FormBuilder refactoring in this PR should go through no matter which plugin chooses to overwrite a view. (The plugin then still has some shitty view code, but the user shouldn't see a difference).

However, the original issue (required fields are not always marked with `*`) might still persist for people using those plugins. Still, this should not affect this PR. You might want to add some notes to the affected work packages that the work packaged should not be marked as 'fixed' until the plugins are fixed too.

**Comment by sschu** [comment]
Well, what I am actually trying to do is create more work for you... The original intent was to fix certain accessibility problems because somebody wants them to be fixed. Unfortunately, that somebody uses those plugins so the problems are really in the plugins. I just think it would be better to fix all of them together by the one who knows how to fix them ... which is ... tada ... you. :)

File: config/routes.rb (hunk 1 above)
Comment (tessi) [inline config/routes.rb:121]: There are tests now

**Comment by tessi** [comment]
@ulferts the required_field_name helper is now used in app/views/planning_element_type_colors/_form.html.erb

**Comment by tessi** [comment]
The following plugins need refinements:
- [x] mobile_otp
- [x] self_registration_by_mail
- [x] dtag_authentication
- [x] two_person_registration

**Comment by tessi** [comment]
Someone may review this PR again. Please also review the pull requests in the mentioned plugins - they need to be merged together.

**Comment by ulferts** [comment]
Never thought about it and I have never seen it anywhere but would our login/password fields on the login form also need the \* marker as they are required? I opt for not adding it as it would be unusual but it is inconsistent.

**Comment by ulferts** [comment]
There is a [minor issue](https://github.com/finnlabs/openproject-two_person_registration/pull/6#issuecomment-32262870) in a plugin. The rest is looking good.

**Comment by tessi** [comment]
And again this PR cannot be merged. We really need to do something against this CHANGELOG-conflic hell

**Pull request merged**